src/tests/ftest/deployment/disk_failure.py

```diff
@@ -0,0 +1,186 @@
+            if value['storage']['smd_info']['devices']:
+                for target in range(self.targets):
+                    device_info[count] = {}
+                    uuid_value = value['storage']['smd_info']['devices'][target]['uuid']
+                    device_info[count]["uuid"] = uuid_value
+                    rank = value['storage']['smd_info']['devices'][target]['rank']
+                    device_info[count]["rank"] = rank
+                    tgt_ids = value['storage']['smd_info']['devices'][target]['tgt_ids']
+                    device_info[count]["tgts"] = tgt_ids
+            count = count + 1
+        return device_info
+
+    def set_nvme_faulty(self, nvme_id=None):
+        """Get a device to faulty state.
+
+        Args:
+            device_id (str): Device UUID
+        Returns:
+            resp (json format) : dmg device faulty information.
+        """
+        data = {}
+        if nvme_id is None:
+            self.fail("No device id provided")
+        self.dmg_command.json.value = True
+        try:
+            result = self.dmg_command.storage_set_faulty(uuid=nvme_id)
+        except process.CmdError as details:
+            self.fail("dmg command failed: {}".format(details))
+        finally:
+            self.dmg_command.json.value = False
+        data = json.loads(result.stdout_text)
+        resp = data['response']
+        return resp
+
+    def verify_disk_failure(self, num_pool):
+        """Run IOR and create disk failures while IO is happening.
+
+        Args:
+            num_pool : Total number of pools to run the testing.
+        """
+        device_info = {}
+        pool = {}
+
+        # Get the device information.
+        device_info = self.get_nvme_device_info()
+        self.log.info("Device information")
+        self.log.info("==================")
+        self.log.info(device_info)
+        for val in range(0, num_pool):
+            pool[val] = add_pool(self, connect=False)
+
+        for val in range(0, num_pool):
+            threads = []
+            self.pool = pool[val]
+            # The following thread runs while raising disk faults
+            threads.append(threading.Thread(target=self.run_ior_thread,
+                                            kwargs={"action": "Write",
+                                                    "oclass": self.ior_cmd.dfs_oclass.value,
+                                                    "test": self.ior_test_sequence[0],
+                                                    "fail_on_warning": False}))
+            # Launch the IOR threads
+            for thrd in threads:
+                self.log.info("Thread : %s", thrd)
+                thrd.start()
+                time.sleep(5)
+
+            count = 0
+            # Evict some of the targets from the system
+            for key, _ in device_info.items():
+                if count < 1:
+                    resp = self.set_nvme_faulty(device_info[key]["uuid"])
```

Comment:
Recommend just using the value from `.items()`
```suggestion
            for key, val in device_info.items():
                if count < 1:
                    resp = self.set_nvme_faulty(val["uuid"])
```

Reply:
Good suggestion. Will do...